Detectors/CTP/reconstruction/include/CTPReconstruction/RawDataDecoder.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file RawDataDecoder.h
+/// \brief Digits  tw Raw translation
+/// \author Roman Lietava
+
+#ifndef ALICEO2_CTP_RAWDATADECODER_H_
+#define ALICEO2_CTP_RAWDATADECODER_H_
+
+#include <vector>
+#include <map>
+#include <deque>
+#include "DPLUtils/DPLRawParser.h"
+#include "Framework/InputRecord.h"
+#include "Framework/InputRecord.h"
+#include "DataFormatsCTP/Digits.h"
+#include "DataFormatsCTP/LumiInfo.h"
+#include "DataFormatsCTP/TriggerOffsetsParam.h"
+
+namespace o2
+{
+namespace ctp
+{
+class RawDataDecoder
+{
+ public:
+  RawDataDecoder() = default;
+  ~RawDataDecoder() = default;
+  static void makeGBTWordInverse(std::vector<gbtword80_t>& diglets, gbtword80_t& GBTWord, gbtword80_t& remnant, uint32_t& size_gbt, uint32_t Npld);
+  int addCTPDigit(uint32_t linkCRU, uint32_t triggerOrbit, gbtword80_t& diglet, gbtword80_t& pldmask, std::map<o2::InteractionRecord, CTPDigit>& digits);
+  int decodeRaw(o2::framework::InputRecord& inputs, std::map<o2::InteractionRecord, CTPDigit>& digits, std::vector<LumiInfo>& lumiPointsHBF1);
+  void setDoLumi(bool lumi) { mDoLumi = lumi; }
+  void setDoDigits(bool digi) { mDoDigits = digi; }
+  void setVerbose(bool v) { mVerbose = v; }
+  uint32_t getIRRejected() { return mIRRejected; }
+  uint32_t getTCRRejected() { return mTCRRejected; }
+  std::vector<uint32_t> getTFOrbits() { return mTFOrbits; }
```

Comment:
Here you return a copy of the vector but as far as I can see you never modify it. I guess it can be
```
const std::vector<uint32_t>& getTFOrbits() const { return mTFOrbits; }
```